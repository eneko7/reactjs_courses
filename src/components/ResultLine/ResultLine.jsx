import React from 'react';
import propTypes from 'prop-types';
import styles from './ResultLine.scss';
import SortBy from './SortBy';

const ResultLine = ({
  quantity,
  searchBy,
  searchRequest,
  location: { pathname },
}) => (
  <div className={styles.result_line}>
    <span
      className={styles.result_line__count_founded_films}
    >
      {`${quantity} movies found ${searchRequest ? `by ${searchBy}: ${searchRequest}` : ''}`}
    </span>
    {pathname.substring(0, 7) !== '/movie/' && <SortBy /> }
  </div>
);

ResultLine.defaultProps = {
  quantity: 1,
  searchRequest: '',
};

ResultLine.propTypes = {
  location: propTypes.shape({
    pathname: propTypes.string,
  }).isRequired,
  quantity: propTypes.number,
  searchBy: propTypes.string.isRequired,
  searchRequest: propTypes.string,
};

export default ResultLine;
