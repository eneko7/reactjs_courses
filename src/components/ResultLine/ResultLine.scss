@import "../../assets/constants/constants.scss";

.result_line {
  padding: 1% 15%;
  background-color: #f2f2f2;
  display: flex;
  justify-content: space-between;

  .result_line__count_founded_films {
    font-family: $fontFamilyBold;
    font-size: 20px;
  }
}

@media (max-width: $fulhdDesktop) {
  .result_line {
    padding: 1% $leftPaddingLaptop;
  }
}

@media (max-width: $laptop) {
  .result_line {
    padding: 1% $leftPaddingTablet;
  }
}

@media (max-width: $tablet) {
  .result_line {
    padding: 1% 5%;
    display: block;
    text-align: right;

    .result_line__count_founded_films {
      font-size: 18px;
    }
  }
}
