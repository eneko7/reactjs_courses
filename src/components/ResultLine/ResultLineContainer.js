import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import ResultLine from './ResultLine';
import { fetchDefaultMovies } from '../../modules/movies/moviesActions';
import { getMoviesState, getQuantityMoviesState } from '../../modules/movies/moviesSelectors';
import { getSearchBy, getSearchRequest } from '../../modules/filter/filterSelectors';

const mapStateToProps = state => ({
  movies: getMoviesState(state),
  quantity: getQuantityMoviesState(state),
  searchBy: getSearchBy(state),
  searchRequest: getSearchRequest(state),
});

const mapDispatchToProps = {
  fetchDefaultMovies,
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ResultLine));
