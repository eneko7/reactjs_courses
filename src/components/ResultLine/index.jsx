import ResultLine from './ResultLineContainer';

export default ResultLine;
