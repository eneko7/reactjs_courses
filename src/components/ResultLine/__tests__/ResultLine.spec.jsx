import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ResultLine from '../ResultLine';

const renderer = new ShallowRenderer();
renderer.render(<ResultLine />);
describe('ResultLine', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
