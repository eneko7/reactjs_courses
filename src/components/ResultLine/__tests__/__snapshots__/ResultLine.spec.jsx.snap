// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultLine renders correctly 1`] = `
<div
  className="result_line"
>
  <span
    className="result_line__count_founded_films"
  >
    1 movies found 
  </span>
  <div
    className="sort_by"
  >
    <span
      className="sort_by__span"
    >
      Sort by
    </span>
    <div>
      test
    </div>
    <div>
      test
    </div>
  </div>
</div>
`;

exports[`ResultLine renders correctly 2`] = `
<div
  className="result_line"
>
  <span
    className="result_line__count_founded_films"
  >
    1 movies found by title: test
  </span>
  <div
    className="sort_by"
  >
    <span
      className="sort_by__span"
    >
      Sort by
    </span>
    <div>
      test
    </div>
    <div>
      test
    </div>
  </div>
</div>
`;
